examples/basic-chat-name.html

```diff
@@ -0,0 +1,266 @@
+      NAF.schemas.getComponentsOriginal = NAF.schemas.getComponents;
+      NAF.schemas.getComponents = (template) => {
+        if (!NAF.schemas.hasTemplate('#avatar-template')) {
+          NAF.schemas.add({
+            template: '#avatar-template',
+            components: [
+              // position and rotation are added by default if we don't include a template, but since
+              // we also want to sync the color, we need to specify a custom template; if we didn't
+              // include position and rotation in this custom template, they'd not be synced.
+              'position',
+              'rotation',
+
+              // this is how we sync a particular property of a particular component for a particular
+              // child element of template instances.
+              {
+                selector: '.head',
+                component: 'material',
+                property: 'color' // property is optional; if excluded, syncs everything in the component schema
+              }
+            ]
+          });
+        }
+        const components = NAF.schemas.getComponentsOriginal(template);
+        return components;
+      };
+
+
+
+document.addEventListener("DOMContentLoaded", () => {
+
+    
+    
+    
+    const btn = document.querySelector("#send-btn");      // SEND btn
+    const input = document.querySelector("#input");     // input field with the text
+    const log = document.querySelector(".messages")    // message log
+    const username = document.querySelector("#user-name");
+username.value = "user-" + Math.round(Math.random() * 10000)
+    
+
+    // when you want to send a message
+    btn.addEventListener("click", evt => {
+        
+        
+
+
+        // log your own messages
+        log.innerHTML += username.value + ": " + input.value + '<br>'
+        // broadcast the text as some unique dataType (like "chat")
+        NAF.connection.broadcastData("chat", {txt: input.value ,
+        // Assign the name as username
+            name: username.value}) 
+
+
+// console.log(NAF.clientId , input.value)
+
+    })
+    
+    // when a "chat" type message arrives
+    NAF.connection.subscribeToDataChannel("chat", (senderId, dataType, data, targetId) => {
+        // append the data.txt to the message log and data.name as username
+        log.innerHTML += data.name + ": " + data.txt + '<br>'
+    })
+    
+})
+
+
+
+    </script>
+    <script src="https://unpkg.com/aframe-randomizer-components@^3.0.1/dist/aframe-randomizer-components.min.js"></script>
+  </head>
+  <body>
+
+
+
+
+
+    <div class="chat">
+        <div style="display: flex;">
+            <input type="text" name="" id="input">
+            <button id="send-btn">Send</button>
+        </div>
+        <div class="messages">
+    
+        </div>
+    </div>
+    
+
+    <div style="z-index: 100; bottom: 24px; left: 48px; position: fixed;">
+<h3 style="color: #fff;">Your Name :</h3>
```

Comment:
You can add "user-select: none" in the style here to avoid the text to be selected when you drag the mouse.